sstables/sstables.cc

```diff
@@ -1872,6 +1875,15 @@ sstable::write_scylla_metadata(shard_id shard, sstable_enabled_features features
         _components->scylla_metadata->data.set<scylla_metadata_type::ExtTimestampStats>(std::move(*ts_stats));
     }
 
+    sstable_id sid;
+    if (generation().is_uuid_based()) {
+        sid = sstable_id(generation().as_uuid());
+    } else {
+        sid = sstable_id(utils::UUID_gen::get_time_UUID());
+        sstlog.info("SSTable generation {} is numerical. SSTable identifier in scylla_metadata set to {}", generation(), sid);
```

Comment:
To be useful, this log message should contain enough information about the sstable, so that we can identify it. Best to include its path in the message.

Reply:
Done in [1f1c797](https://github.com/scylladb/scylladb/commit/1f1c7972194188d8f5cf96ee249bb5ba4d1da9b2)